src/supermarket/app/controllers/api/v1/cookbook_versions_controller.rb

```diff
@@ -112,4 +117,13 @@ def require_collaborator_params
     params.require(:collaborator_failure)
     params.require(:collaborator_feedback)
   end
+
+  def create_metric(cookbook_version, quality_metric, failure, feedback)
+    MetricResult.create!(
+      cookbook_version: cookbook_version,
+      quality_metric: quality_metric,
+      failure: failure,
+      feedback: feedback
+    )
+  end
```

Comment:
OK. By my reading, the duplication prevented is not having to type the keyword arguments which I suppose if we suspect they might change in future, this way we've got fewer places to update them.